bc_obps/registration/api/operation.py

```diff
@@ -287,10 +287,16 @@ def update_operation(request, operation_id: int, submit: str, payload: Operation
             operator.set_archive(modifier=user)
 
     if payload.statutory_declaration:
+        operation_document = Operation.documents.through.objects.filter(
+            operation_id=operation.id
+        ).first()  # there will only ever be 1 statutory declaration
+        if operation_document:
+            Document.objects.get(
+                id=operation_document.document_id
+            ).delete()  # we delete here rather than update because delete removes the file from google cloud storage
```

Comment:
To simplify this a little we could do:
```python
operation.documents.filter(type=DocumentType.objects.get(name="signed_statutory_declaration")).delete()
```
If there will only ever be 1 statutory declaration.